### ISSUE TITLE
`no-unreachable-types` doesn't seem to work properly when Query/Mutation types are defined in a separate file

### ISSUE THREAD

**Issue by Berger92** [B2o5T/graphql-eslint#1126]
### Issue workflow progress

_Progress of the issue based on the [Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [ ] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox
> Please make sure the graphql-eslint version under `package.json` matches yours.
- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**

Hi 👋 

We're trying to use this library to lint some of our subgraphs (in an Apollo Federation) and as per the sandbox below, we are getting an error for all of our types from `no-unreachable-types` rule. Are we doing something wrong, or is there a real issue here? 🤔 

Thanks.

**To Reproduce**

[CodeSandbox](https://codesandbox.io/s/unreachable-types-issue-ljup2q?file=/src/subgraph.graphql)
Run `yarn lint` in the console

**Expected behavior**

`Book` is reachable and it shouldn't throw an Error for it.

**Environment:**

- OS: MacOS
- `@graphql-eslint/eslint-plugin`: 3.10.7
- Node.js: 16.15

**Comment by B2o5T** [comment]
It’s works perfectly fine, provide full schema for graphql-eslint

```diff
parserOptions:
  schema:
    - './src/schema.graphql'
+   - './src/subgraph.graphql'
```

**Comment by Berger92** [comment]
@B2o5T Thanks for the quick response. I've tried your proposed solution in the sandbox, but it doesn't work. Did I do something wrong again? 🤔 
I've updated the sanbox.
<img width="1458" alt="image" src="https://user-images.githubusercontent.com/8849858/183284294-96190bdd-71e1-49b6-9b5a-9a81c1de5292.png">
